packages/core/src/lib/relay/index.ts

```diff
@@ -182,24 +213,70 @@ class Relay extends GossipSub implements IRelay {
       }
     );
 
-    this.topicValidators.set(pubSubTopic, messageValidator);
-    super.subscribe(pubSubTopic);
+    this.gossipSub.topicValidators.set(pubSubTopic, messageValidator);
+    this.gossipSub.subscribe(pubSubTopic);
   }
 
-  unsubscribe(pubSubTopic: TopicStr): void {
-    super.unsubscribe(pubSubTopic);
-    this.topicValidators.delete(pubSubTopic);
-  }
-
-  getMeshPeers(topic?: TopicStr): PeerIdStr[] {
-    return super.getMeshPeers(topic ?? this.pubSubTopic);
+  private isRelayPubSub(pubsub: PubSub): boolean {
+    return pubsub?.multicodecs?.includes(Relay.multicodec) || false;
   }
 }
 
-Relay.multicodec = constants.RelayCodecs[constants.RelayCodecs.length - 1];
-
 export function wakuRelay(
+  init: Partial<ProtocolCreateOptions> = {}
+): (libp2p: Libp2p) => IRelay {
+  return (libp2p: Libp2p) => new Relay(libp2p, init);
+}
+
+export function wakuGossipSub(
```

Comment:
This is a consequence of `gossipSub` being a property and not an extension of the `Relay`.
My opinion - it makes sense because `Relay` is our abstraction and `GossipSub` is a tool that we use to implement what we need. 

cc: @fryorcraken

Reply:
I see how it is. I think it's fine for now. We can iterate on the API as we use it and get more feedback.